src/main/resources/js/colorPickMacro.js

```diff
@@ -0,0 +1,91 @@
+const baseUrl = AJS.params.baseUrl;
+const pageID = AJS.params.pageId;
+
+function updateImage(file, uniquefilename) {
+    file.name = uniquefilename;
+    console.log(file);
+    var actionData = new FormData();
+    actionData.append('file', file, uniquefilename);
+    actionData.append('comment', "foobar");
+    actionData.append('minorEdit', "true");
+    AJS.$.ajax({
+        url: baseUrl + '/rest/api/content/' + pageID + '/child/attachment',
+        type: "POST",
+        data: actionData,
+        dataType: "json",
+        processData: false,
+        headers: {
+            "X-Atlassian-Token": "nocheck",
+        },
+        contentType: false,
+        cache: false,
+        success: [function (content) {
```

Comment:
you don't need an array of functions, because you use only one function, same for the error and other ajax queries below.